File: examples/HelloPeople/python/bin/hellopeople.py
#!/usr/bin/env python2
import sys

from dataclay.api import init, finish

# Init dataClay session
init()

from HelloPeople_ns.classes import Person, People


class Attributes(object):
    pass


def usage():
    print("Usage: hellopeople.py <colname> <personName> <personAge>")


def init_attributes(attributes):
    if len(sys.argv) != 4:
        print("ERROR: Missing parameters")
        usage()
        exit(2)
    attributes.collection = sys.argv[1]
    attributes.p_name = sys.argv[2]
    attributes.p_age = int(sys.argv[3])


if __name__ == "__main__":
    attributes = Attributes()
    init_attributes(attributes)

    # Create people object if it does not exist
    try:
        # Trying to retrieve it using alias
        people = People.get_by_alias(attributes.collection)
    except:
        print("[LOG] Creating people's object with alias %s" \
            % attributes.collection)
        people = People()
        people.make_persistent(alias=attributes.collection)

    # Add new person to people
    person = Person(attributes.p_name, attributes.p_age)
    person.make_persistent()
    people.add(person)

    print("[LOG] People object contains:")
    print(people)

    # Close session
    finish()
    exit(0)

File: examples/HelloPeople/java/src/app/HelloPeople.java
package app;

import dataclay.api.DataClay;
import model.People;
import model.Person;

public class HelloPeople {
	private static void usage() {
		System.out.println("Usage: application.HelloPeople <peopleAlias> <personName> <personAge>");
		System.exit(1);
	}

	public static void main(final String[] args) {
		try {
			// Check and parse arguments
			if (args.length != 3) {
				usage();
			}
			final String peopleAlias = args[0];
			final String pName = args[1];
			final int pAge = Integer.parseInt(args[2]);

			// Init dataClay session
			DataClay.init();

			// Access (or create collection)
			People people = null;
			try {
				people = People.getByAlias(peopleAlias);
				System.out.println("[LOG] People object " + peopleAlias + " found.");
			} catch (final Exception ex) {
				System.out.println("[LOG] Creating NEW object People");
				people = new People();
				people.makePersistent(peopleAlias);
			}

			// Create person
			final Person person = new Person(pName, pAge);
			person.makePersistent();
			people.add(person);

			// Print people (people iterated remotely)
			System.out.println(people);

			// Finish dataClay session
			DataClay.finish();

			// Exit
			System.exit(0);
		} catch (final Exception e) {
			e.printStackTrace();
			System.exit(1);
		}
	}
}
